packages-content-model/roosterjs-content-model-core/lib/modelApi/edit/deleteExpandedSelection.ts

```diff
@@ -128,6 +130,10 @@ export function deleteExpandedSelection(
     return context;
 }
 
+function getMarkerFormat(segment: ContentModelSegment): ContentModelSegmentFormat | undefined {
+    return segment.segmentType !== 'Image' ? segment.format : undefined;
```

Comment:
Instead of remove all format from imageFormat, we should keep those format for text.

I have a function added to do this but the code is not ready for review yet. You can refactor it and add in your PR now and make it public.

https://github.com/microsoft/roosterjs/blob/ea972ffb8449e828fd071658c52b787043b92a67/packages-content-model/roosterjs-content-model-core/lib/publicApi/model/paste.ts#L112